### PR TITLE
AISDK-115: Add remove_disfluencies to Java sdk

### PR DESCRIPTION
**Pull request by timitijani**
* Added remove_disfluencies option to Async and Streaming clients
* Updated examples to use the new option
* Updated tests (unit + integration)

### CODE REVIEW

**Comment by timitijani** [comment]
Bloated by running `Refactor Code` which was mentioned in README

File: src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java [hunk 1]
```diff
@@ -27,103 +27,115 @@
...
+                        {METADATA, VOCAB_ID, true, null},
+                        {METADATA, VOCAB_ID, false, null},
+                        {METADATA, VOCAB_ID, null, null},
+                        {METADATA, null, true, true},
+                        {null, VOCAB_ID, true, false},
+                        {null, VOCAB_ID, false, null},
+                        {null, VOCAB_ID, null, true},
+                        {null, null, true, true},
+                        {null, null, false, false},
+                        {null, null, null, null}
```
Comment (KostasRev) [inline src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java]: Please check if there's another way to specify these option sets, this matrix is not sustainable. If possible we should change it starting from this PR.

**Review by KostasRev — CHANGES_REQUESTED** [review]
In addition to the docs change we mentioned, please see the parametrization comment, but otherwise looks good!

Make sure to locally test against API after making all the changes.

File: src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java (hunk 1 above)
Comment (amikofalvy) [reply src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java]: ☝️ I agree

Comment (timitijani) [reply src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java]: I changed it to `@Theory` and it tests all the cases.

**Review by amikofalvy — APPROVED** [review]
Please verify that a broken test case using theories fails the test and shows up as a failure in the test suite. Afterward, :shipit:

File: pom.xml [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <modelVersion>4.0.0</modelVersion>
     <groupId>ai.rev.speechtotext</groupId>
     <artifactId>revai-java-sdk-speechtotext</artifactId>
-    <version>1.0.0</version>
+    <version>1.1.0</version>
```
Comment (KostasRev) [inline pom.xml:7]: Please double check with @Chanc3 that this is the proper way to bump the version, but seems reasonable to me.

File: src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java [hunk 3]
```diff
@@ -10,120 +10,125 @@
...
 import java.io.UnsupportedEncodingException;
 import java.net.URLDecoder;
-import java.util.Arrays;
-import java.util.Collection;
 import java.util.concurrent.TimeUnit;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-@RunWith(Parameterized.class)
+@RunWith(Theories.class)
```
Comment (KostasRev) [inline src/test/java/ai/rev/speechtotext/unit/RevAiStreamingClientOptionalParametersTest.java:25]: Fantastic, much better now!

**Review by KostasRev — APPROVED** [review]
Looking much better with the theory now! Nice work, please address Andrew's suggestion and 🚢

File: pom.xml (hunk 2 above)
Comment (timitijani) [reply pom.xml:7]: Talked to @Chanc3 , this is the correct way to bump the version.

**Pull request merged**